src/main/java/com/example/naitei19javaecommerce/service/Impl/InvoiceServiceImpl.java

```diff
@@ -38,6 +37,21 @@ public InvoiceDTO findInvoiceById(Long id) {
         return null;
     }
 
+    @Override
+    public List<InvoiceDTO> findInvoicesListByUserId(Long userId) {
+        List<Invoice> invoices = invoiceRepository.findProductRelatedById(userId);
+        if (invoices.isEmpty()){
+            return null;
+        }
+        List<InvoiceDTO> invoiceDTOs = new ArrayList<>();
+        for (Invoice invoice : invoices) {
+            List<InvoiceDetailDTO> invoiceDetails = invoiceDetailRepository.findInvoiceDetailsByProductId(invoice.getId())
+                    .stream().map((invoiceDetail) -> mapToInvoiceDetailDto(invoiceDetail)).collect(Collectors.toList());
+            invoiceDTOs.add(mapToInvoiceDto(invoice, invoiceDetails));
+        }
```

Comment:
thay vì for thế này, em dùng eager để load invoice và load cả invoice deails tương ứng luôn
-> dùng type select nha